packages/core/src/ui/widgets/colorBar/ColorBarScale.ts

```diff
@@ -0,0 +1,398 @@
+    }
+
+    this._range = range;
+    this.render();
+  }
+
+  public get voiRange() {
+    return { ...this._voiRange };
+  }
+
+  public set voiRange(voiRange: ColorBarVOIRange) {
+    if (!isRangeValid(voiRange) || rangesEqual(voiRange, this._voiRange)) {
+      return;
+    }
+
+    this._voiRange = voiRange;
+    this.render();
+  }
+
+  public get tickSize(): number {
+    return this._tickSize;
+  }
+
+  public set tickSize(tickSize: number) {
+    if (tickSize === this._tickSize) {
+      return;
+    }
+
+    this._tickSize = tickSize;
+    this.render();
+  }
+
+  public get tickWidth(): number {
+    return this._tickWidth;
+  }
+
+  public set tickWidth(tickWidth: number) {
+    if (tickWidth === this._tickWidth) {
+      return;
+    }
+
+    this._tickWidth = tickWidth;
+    this.render();
+  }
+
+  public get tickColor(): string {
+    return this._color;
+  }
+
+  public set tickColor(tickColor: string) {
+    if (tickColor === this._color) {
+      return;
+    }
+
+    this._color = tickColor;
+    this.render();
+  }
+
+  public get showFullPixelValueRange(): boolean {
+    return this._showFullPixelValueRange;
+  }
+
+  public set showFullPixelValueRange(showFullRange: boolean) {
+    if (showFullRange === this._showFullPixelValueRange) {
+      return;
+    }
+
+    this._showFullPixelValueRange = showFullRange;
+    this.render();
+  }
+
+  public get visible() {
+    return this._canvas.style.display === 'block';
+  }
+
+  public set visible(visible) {
+    if (visible === this.visible) {
+      return;
+    }
+
+    this._canvas.style.display = visible ? 'block' : 'none';
+
+    if (visible) {
+      this.render();
+    }
+  }
+
+  public appendTo(container: HTMLElement) {
+    container.appendChild(this._canvas);
+    this.render();
+  }
+
+  private static validateProps(props: ColorBarScaleProps) {
+    const { size, range, voiRange } = props;
+
+    if (size && !isSizeValid(size)) {
+      throw new Error('Invalid "size"');
+    }
+
+    if (range && !isRangeValid(range)) {
+      throw new Error('Invalid "range"');
+    }
+
+    if (voiRange && !isRangeValid(voiRange)) {
+      throw new Error('Invalid "voiRange"');
+    }
+  }
+
+  private _setCanvasSize(canvas: HTMLCanvasElement, size: ColorBarSize) {
+    const { width, height } = size;
+
+    canvas.width = width;
+    canvas.height = height;
+
+    Object.assign(canvas.style, {
+      width: `${width}px`,
+      height: `${height}px`,
+    });
+  }
+
+  private _getCanvasPosition(canvas): ColorBarPosition {
+    const { top: canvasTop, left: canvasLeft } = this._canvas.style;
+    const top = Number.parseInt(canvasTop);
+    const left = Number.parseInt(canvasLeft);
+
+    return { top, left };
+  }
+
+  private _setCanvasPosition(
+    canvas: HTMLCanvasElement,
+    position: ColorBarPosition
+  ) {
+    Object.assign(canvas.style, {
+      top: `${position.top}px`,
+      left: `${position.left}px`,
+    });
+  }
+
+  private _createCanvasElement(
```

Comment:
hmmm so the ticks are their own canvas and the colorbar is its own? I think that makes sense